eng/pipelines/templates/jobs/ci.yml

```diff
@@ -275,16 +276,18 @@ jobs:
           AdditionalModules: ${{ parameters.AdditionalModules }}
           JobType: 'Analyze'
 
-      - task: PythonScript@0
-        displayName: 'Generate directories variable for sparse checkout'
+      - task: PowerShell@2
+        displayName: Generate directories variable for sparse checkout
         inputs:
-          scriptPath: 'eng/scripts/generate_from_source_pom.py'
-          arguments: '--set-pipeline-variable CheckoutDirectories --project-list $(ProjectList)'
-          workingDirectory: '$(System.DefaultWorkingDirectory)'
+          pwsh: true
+          filePath: $(Build.SourcesDirectory)/eng/scripts/Generate-ServiceDirectories-From-Project-List.ps1
+          arguments: >
+            -SourcesDirectory $(Build.SourcesDirectory)
+            -ProjectList $(ProjectList)
 
       - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
         parameters:
-          Paths: $(CheckoutDirectories)
+          Paths: $(SparseCheckoutDirectories)
```

Comment:
nit: I was trying to figure out where this is set and until I reviewed the script it wasn't clear that is what set it. In other cases we have made that a parameter into the script so things were a little more connected. It looks like you are setting 2 variables which is probably why you opted not to do that.

Reply:
Previously they were passing this into the python script and calling it prior to every sparse checkout that wasn't the default. There's no real point in passing in an environment variable to set if it's always being set in multiple places. Further, I'd added a second variable for ServiceDirectires which isn't yet being used but will be so I saw no point in passing in variables that were being expected.